app/Exceptions/Handler.php

```diff
@@ -153,51 +153,55 @@ public function render($request, \Throwable $e): JsonResponse|Response
                     return response()
                         ->json([
                             'status' => $e->getCode(),
-                            'type' => 'BadResponseException',
-                            'message' => 'Jikan failed to connect to MyAnimeList.net. MyAnimeList.net may be down/unavailable, refuses to connect or took too long to respond.',
+                            'type' => 'UpstreamException',
+                            'message' => 'Request to MyAnimeList.net failed. MyAnimeList.net may be down/unavailable, refuses to connect or took too long to respond. Please try again later.',
                             'error' => $e->getMessage()
-                        ], 503);
+                        ], 504);
```

Comment:
This should be 500. In terms of REST API and the client's request to Jikan API, 504 doesn't make sense, because the client's request was received by the app. The timeout happened with an upstream service, and all http apps return 500 in those cases, that's the standard.